### DISCUSSION TITLE
Conversion soap to graphql

### DISCUSSION THREAD

**Discussion by PractiseGraphql** [community/community#128241] [npm]
Hello,

I am trying to explore ways for converting soap wsdl to graphql using npm soap-graphql package.Its really helpful if any reference document or some steps will be provided.I can follow the same and will start exploring more.

**Comment by bernardo-amaral** [comment]
its better convert your application manually... automatic conversions will broke yout app. good luck in the conversion!

**Comment by PractiseGraphql** [comment]
Hi ,
I tried by doing manual way .with this wsdl(http://webservices.oorsprong.org/websamples.countryinfo/CountryInfoService.wso?WSDL) its working ,able to execute and its giving response also.
But , with this one (https://flexera1000-ci.flexnetoperations.com/flexnet/services/TestService?wsdl) getting response as null
<img width="949" alt="image" src="https://github.com/community/community/assets/170598356/8da63403-fa70-4dfd-9dca-06c93c2676b2">

By giving same inputs in SOAP getting response but from graphql not getting any response
<img width="598" alt="image" src="https://github.com/community/community/assets/170598356/1ed661f1-e731-44d3-9261-0e9225eae3b8">


This is my server.js file-
`import express from 'express';
import { graphqlHTTP } from 'express-graphql';
import { buildSchema } from 'graphql';
import soap from 'soap';


const app = express();

// SOAP WSDL URL
const wsdlUrl = 'https://flexera1000-ci.flexnetoperations.com/flexnet/services/TestService?wsdl';

// Create SOAP Client
soap.createClient(wsdlUrl, (err, soapClient) => {
  if (err) throw err;

  // Define GraphQL Schema
  const schema = buildSchema(`
    type Query {
      createTestRequest(name: String, intCount: String): String
    }
  `);

  // Define Resolvers
  const root = {
    createTest: async ({ name,intCount }) => {
      return new Promise((resolve, reject) => {
        soapClient.createTest({ name: name, intCount: intCount }, (err, result) => {
          if (err) reject(err);
          resolve(result.createTestResponse);
        });
      });
    },
  };

  // Use GraphQL Middleware
  app.use('/graphql', graphqlHTTP({
    schema: schema,
    rootValue: root,
    graphiql: true,
  }));

  // Start Server
  app.listen(4000, () => {
    console.log('Server is running on http://localhost:4000/graphql');
  });
})`

This is my package.json file:

`{
  "name": "soap-graphql",
  "version": "1.0.0",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "keywords": [],
  "description": "",
  "dependencies": {
    "express": "^4.19.2",
    "express-graphql": "^0.12.0",
    "graphql": "^16.8.2",
    "node-soap": "^1.0.0",
    "soap": "^1.0.3",
    "soap-graphql": "^1.1.0"
  }
}
`

**Reply by nane-petrosyan** [reply]
Hello, you have a resolver for `createTest` but your schema contains a query called `createTestRequest`

**Comment by Zobb0908** [comment]
Hi I this tool may help you in future [https://chatgtpai.org](url)
